### PR TITLE
literal: round-half-even tie for normal-range float/complex repr

### PR DESCRIPTION
**Pull request by youknowone**
`float::to_string`'s normal-magnitude branch and `complex::component_to_string` returned Rust's shortest formatting directly. Rust's formatter can land on the odd-digit neighbour of a rounding tie where `repr()`'s round-half-to-even picks the even one, e.g. bits `0x42e26687db6b9b04` format as `161852602146008.13` but CPython `repr` gives `161852602146008.12` (both round-trip).

The scientific branch of `float::to_string` already ran through `prefer_cpython_tie_repr`; this extends that helper to fixed-notation strings and applies it to the normal-magnitude float branch and to both complex branches.

Verified against CPython 3.14 over a differential battery (20k floats, 280k complex pairs incl. subnormals / powers of ten / ties): **0 divergences** after the fix. Added a normal-range tie unit test.

— *assisted by Claude*

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved `repr()`/`str()` formatting for floating-point values in round-half-to-even tie cases, including normal-range and scientific-notation outputs.
  * Enhanced complex number formatting so real/imaginary components reflect the expected tie digit behavior, including signed zero scenarios.
  * Improved handling of decimal inputs lacking an exponent when generating representations.
* **Tests**
  * Added exact `repr()` assertions for built-in float and complex edge cases covering positive/negative components and tie behavior.

### CODE REVIEW

File: crates/literal/src/float.rs [hunk 1]
```diff
@@ -351,6 +353,21 @@ mod tests {
             "6.1005353927612305e-05"
         );
     }
+
+    #[test]
+    fn repr_normal_range_uses_cpython_tie_digit() {
```
Comment (youknowone) [inline crates/literal/src/float.rs:358]: Add also extra_tests
